### PR TITLE
DAOS-7708 tests: Updating server/metadata.py test container max quantity

### PR DESCRIPTION
**Pull request by phender**
Updating the server/metadata.py tests to dynamically set the maximum
number of containers that can be created.

Also resolving DAOS-7710 by enabling the test_metadata_server_restart in
weekly CI execution.

Quick-Functional: true
Test-tag: metadata

Signed-off-by: Phillip Henderson <phillip.henderson@intel.com>

### CODE REVIEW

**Comment by phender** [comment]
All 4 of the server/metadata.py tests passed in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6272/22/artifact/Functional%20Hardware%20Large/server/metadata.py/results.html

**Comment by phender** [comment]
Results from the updated `server/metadata.py:ObjectMetadata.test_metadata_addremove` test:
```
2021-08-31 17:55:58,245 metadata         L0314 INFO | Summary
2021-08-31 17:55:58,245 metadata         L0315 INFO |   Loop  Containers Created  Variation
2021-08-31 17:55:58,245 metadata         L0316 INFO |   ----  ------------------  ---------
2021-08-31 17:55:58,245 metadata         L0333 INFO |   1     3245                0
2021-08-31 17:55:58,245 metadata         L0333 INFO |   2     3210                -35
2021-08-31 17:55:58,246 metadata         L0333 INFO |   3     3205                -5
2021-08-31 17:55:58,246 metadata         L0333 INFO |   4     3244                39
2021-08-31 17:55:58,246 metadata         L0333 INFO |   5     3225                -19
2021-08-31 17:55:58,246 metadata         L0333 INFO |   6     3229                4
2021-08-31 17:55:58,246 metadata         L0333 INFO |   7     3229                0
2021-08-31 17:55:58,246 metadata         L0333 INFO |   8     3244                15
2021-08-31 17:55:58,246 metadata         L0333 INFO |   9     3224                -20
2021-08-31 17:55:58,246 metadata         L0333 INFO |   10    3228                4
2021-08-31 17:55:58,246 metadata         L0337 INFO | Test passed
```

**Review by shimizukko — DISMISSED** [review]

**Comment by phender** [comment]
Results from https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6272/26/artifact/Functional%20Hardware%20Large/server/metadata.py/job.log:
```
2021-09-03 03:02:37,045 metadata         L0314 INFO | Summary
2021-09-03 03:02:37,045 metadata         L0315 INFO |   Loop  Containers Created  Variation
2021-09-03 03:02:37,046 metadata         L0316 INFO |   ----  ------------------  ---------
2021-09-03 03:02:37,046 metadata         L0334 INFO |   1     3231                0
2021-09-03 03:02:37,046 metadata         L0334 INFO |   2     3229                -2
2021-09-03 03:02:37,046 metadata         L0334 INFO |   3     3244                15
2021-09-03 03:02:37,046 metadata         L0334 INFO |   4     3225                -19
2021-09-03 03:02:37,046 metadata         L0334 INFO |   5     3228                3
2021-09-03 03:02:37,046 metadata         L0334 INFO |   6     3231                3
2021-09-03 03:02:37,046 metadata         L0334 INFO |   7     3230                -1
2021-09-03 03:02:37,046 metadata         L0334 INFO |   8     3230                0
2021-09-03 03:02:37,046 metadata         L0334 INFO |   9     3231                1
2021-09-03 03:02:37,046 metadata         L0334 INFO |   10    3204                -27
2021-09-03 03:02:37,046 metadata         L0338 INFO | Test passed
```

**Review by shimizukko — APPROVED** [review]

File: src/tests/ftest/server/metadata.py [hunk 1]
```diff
@@ -306,78 +424,65 @@ def test_metadata_server_restart(self):
...
 
-    @avocado.fail_on(DaosApiError)
+        self.log.info("Test passed")
+
     def test_container_removal_after_der_nospace(self):
-        """JIRA ID: DAOS-4858
+        """JIRA ID: DAOS-4858.
 
         Test Description:
            Verify container can be successfully deleted when the storage pool
```
Comment (kccain) [inline src/tests/ftest/server/metadata.py:433]: This test seems a little strange, does not seem to be doing what is specified in https://daosio.atlassian.net/browse/DAOS-4858 

Test as written seems to exercise metadata capacity in overlapping ways to metadata_fillup and addremove tests. Also the ticket seems to describe exercising regular I/Os on the pool target storage capacity. Versus what is implemented so far is exercising metadata capacity only.

A note for the future, not an issue for this PR.

**Review by kccain — APPROVED** [review]
A few minor suggestions but can proceed to gatekeeper review.

File: src/tests/ftest/server/metadata.py (hunk 1 above)
Comment (phender) [reply src/tests/ftest/server/metadata.py:433]: Created https://daosio.atlassian.net/browse/DAOS-8484 to track this.

**Comment by phender** [comment]
The https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6272/26/ run had one known failure:
- osa/osa_offline_reintegration.py:OSAOfflineReintegration.test_osa_offline_reintegration_with_rf: https://daosio.atlassian.net/browse/DAOS-8231

**Comment by sylviachanoiyee** [comment]
@phender , could you document the test steps of server/metadata.py in the wiki here? Thanks!
https://daosio.atlassian.net/wiki/spaces/DAOS/pages/10988486671/Server+Test+Plan

**Pull request merged**